src/assets/TypeScriptAsset.js

```diff
@@ -0,0 +1,34 @@
+const JSAsset = require('./JSAsset');
+const config = require('../utils/config');
+const localRequire = require('../utils/localRequire');
+
+class TypeScriptAsset extends JSAsset {
+  async transform() {
+    this.ast = await this.parse(this.contents);
+    this.isAstDirty = true;
+  }
+
+  async parse(code) {
+    // require typescript, installed locally in the app
+    let typescript = localRequire('typescript', this.name);
+
+    let transpilerOptions = {
+      compilerOptions: {
+        module: typescript.ModuleKind.CommonJS,
```

Comment:
Does this still support ES2015 modules as well?

Reply:
It's configurable by adding a tsconfig.json file, but i can default it to ES2015 if that is preferred?

Reply:
Sorry, I guess I should ask what this option is for. Does it control parsing behavior or code generation? If codegen, we want commonjs for sure.

Reply:
It's codegen